TST: failures in `test_riemann_zeta_complex` on windows in conda-forge

Due to another issue, rather than waiting for the rc's, I've run some [builds](https://github.com/conda-forge/scipy-feedstock/pull/284) in conda-forge from main. As of today, this leads to test failures on windows:
```
=========================== short test summary info ===========================
FAILED special/tests/test_zeta.py::test_riemann_zeta_complex[(-217.40285743524163+13.992648136816397j)-(-6.012818500554211e+249-1.926943776932387e+250j)-5e-13] - AssertionError: 
Not equal to tolerance rtol=5e-13, atol=0

nan location mismatch:
 ACTUAL: array(nan+nanj)
 DESIRED: array(-6.012819e+249-1.926944e+250j)
FAILED special/tests/test_zeta.py::test_riemann_zeta_complex[(-237.71710702931668+13.992648136816397j)-(-8.823803086106129e+281-5.009074181335139e+281j)-1e-13] - AssertionError: 
Not equal to tolerance rtol=1e-13, atol=0

nan location mismatch:
 ACTUAL: array(nan+nanj)
 DESIRED: array(-8.823803e+281-5.009074e+281j)
FAILED special/tests/test_zeta.py::test_riemann_zeta_complex[(-217.40285743524163-16.147667799398363j)-(-5.111612904844256e+251-4.907132127666742e+250j)-5e-13] - AssertionError: 
Not equal to tolerance rtol=5e-13, atol=0

nan location mismatch:
 ACTUAL: array(nan+nanj)
 DESIRED: array(-5.111613e+251-4.907132e+250j)
FAILED special/tests/test_zeta.py::test_riemann_zeta_complex[(-237.71710702931668-16.147667799398363j)-(-1.3256112779883167e+283-2.253002003455494e+283j)-5e-13] - AssertionError: 
Not equal to tolerance rtol=5e-13, atol=0

nan location mismatch:
 ACTUAL: array(nan+nanj)
 DESIRED: array(-1.325611e+283-2.253002e+283j)
= 4 failed, 72607 passed, 4723 skipped, 326 xfailed, 16 xpassed, 2594 warnings in 845.67s (0:14:05) =
```

CC @steppi for https://github.com/scipy/scipy/pull/21744

For the record, the 4 very large values in the issue description all do indeed underflow, as do 2 others, I noticed from the warnings escaping (that will be fixed) in parallel testing on macOS arm64 with Clang:
```
% pixi r test-nogil -s special -- --parallel-threads=4
....
FAILED scipy/special/tests/test_zeta.py::test_riemann_zeta_complex[(-217.40285743524163+13.992648136816397j)-(-6.012818500554211e+249-1.926943776932387e+250j)-5e-13] - scipy.special._sf_error.SpecialFunctionError: scipy.special/_riemann_zeta: (underflow) floating point underflow
FAILED scipy/special/tests/test_zeta.py::test_riemann_zeta_complex[(-237.71710702931668+13.992648136816397j)-(-8.823803086106129e+281-5.009074181335139e+281j)-1e-13] - scipy.special._sf_error.SpecialFunctionError: scipy.special/_riemann_zeta: (underflow) floating point underflow
FAILED scipy/special/tests/test_zeta.py::test_riemann_zeta_complex[(-294.86605461349745+13.992648136816397j)-(nan+infj)-1e-13] - scipy.special._sf_error.SpecialFunctionError: scipy.special/_riemann_zeta: (underflow) floating point underflow
FAILED scipy/special/tests/test_zeta.py::test_riemann_zeta_complex[(-217.40285743524163-16.147667799398363j)-(-5.111612904844256e+251-4.907132127666742e+250j)-5e-13] - scipy.special._sf_error.SpecialFunctionError: scipy.special/_riemann_zeta: (underflow) floating point underflow
FAILED scipy/special/tests/test_zeta.py::test_riemann_zeta_complex[(-237.71710702931668-16.147667799398363j)-(-1.3256112779883167e+283-2.253002003455494e+283j)-5e-13] - scipy.special._sf_error.SpecialFunctionError: scipy.special/_riemann_zeta: (underflow) floating point underflow
FAILED scipy/special/tests/test_zeta.py::test_riemann_zeta_complex[(-294.86605461349745-16.147667799398363j)-(nan-infj)-1e-13] - scipy.special._sf_error.SpecialFunctionError: scipy.special/_riemann_zeta: (underflow) floating point underflow
```

--- a/scipy/special/tests/test_zeta.py
+++ b/scipy/special/tests/test_zeta.py
@@ -271,7 +271,7 @@ gcc_linux = cpp_compiler == "gcc" and sys.platform == "linux"
 clang_macOS = cpp_compiler == "clang" and sys.platform == "darwin"
 
 
-@pytest.mark.xfail(
+@pytest.mark.skipif(
     not (gcc_linux or clang_macOS),
     reason="Underflow may not be avoided on other platforms",
 )

--- a/scipy/special/tests/test_zeta.py
+++ b/scipy/special/tests/test_zeta.py
@@ -1,5 +1,6 @@
 import scipy
 import scipy.special as sc
+import sys
 import numpy as np
 import pytest
 
@@ -51,11 +52,6 @@ def test_riemann_zeta_avoid_overflow():
     desired = -5.6966307844402683127e+297  # Computed with Mpmath
     assert_allclose(sc.zeta(s), desired, atol=0, rtol=5e-14)
 
-# Some of the test cases below fail for intel compilers
-cpp_compiler = scipy.__config__.CONFIG["Compilers"]["c++"]["name"]
-using_intel_compiler = cpp_compiler in [
-    "intel", "intel-cl", "intel-llvm", "intel-llvm-cl"
-]
 
 @pytest.mark.parametrize(
     "z, desired, rtol",
@@ -261,46 +257,45 @@ using_intel_compiler = cpp_compiler in [
         ((50.55805244181687-1e6j),
          (1.0000000000000002-5.736517078070873e-16j),
          1e-13),
-        # Naive implementation of reflection formula suffers internal overflow
-        pytest.param(
-            (-217.40285743524163+13.992648136816397j),
-            (-6.012818500554211e+249-1.926943776932387e+250j),
-            5e-13,
-            marks=pytest.mark.xfail(
-                condition=using_intel_compiler,
-                reason="Fails with intel compiler.",
-            ),
-        ),
-        pytest.param(
-            (-237.71710702931668+13.992648136816397j),
-            (-8.823803086106129e+281-5.009074181335139e+281j),
-            1e-13,
-            marks=pytest.mark.xfail(
-                condition=using_intel_compiler,
-                reason="Fails with intel compiler.",
-            ),
-        ),
         ((-294.86605461349745+13.992648136816397j), (-np.inf+np.inf*1j), 1e-13),
-        pytest.param(
-            (-217.40285743524163-16.147667799398363j),
-            (-5.111612904844256e+251-4.907132127666742e+250j),
-            5e-13,
-            marks=pytest.mark.xfail(
-                condition=using_intel_compiler,
-                reason="Fails with intel compiler.",
-            ),
-        ),
-        pytest.param(
-            (-237.71710702931668-16.147667799398363j),
-            (-1.3256112779883167e+283-2.253002003455494e+283j),
-            5e-13,
-            marks=pytest.mark.xfail(
-                condition=using_intel_compiler,
-                reason="Fails with intel compiler.",
-            ),
-        ),
         ((-294.86605461349745-16.147667799398363j), (np.inf-np.inf*1j), 1e-13),
     ]
 )
 def test_riemann_zeta_complex(z, desired, rtol):
+    assert_allclose(sc.zeta(z), desired, rtol=rtol)
+
+
+# Some of the test cases below fail for intel compilers
+cpp_compiler = scipy.__config__.CONFIG["Compilers"]["c++"]["name"]
+gcc_linux = cpp_compiler == "gcc" and sys.platform == "linux"
+clang_macOS = cpp_compiler == "clang" and sys.platform == "darwin"
+
+
+@pytest.mark.xfail(
+    not (gcc_linux or clang_macOS),
+    reason="Underflow may not be avoided on other platforms",
+)
+@pytest.mark.parametrize(
+    "z, desired, rtol",
+    [
+        # Test cases generated as part of same script for
+        # test_riemann_zeta_complex. These cases are split off because
+        # they fail on some platforms.
+        #
+        # Naive implementation of reflection formula suffers internal overflow
+        ((-217.40285743524163+13.992648136816397j),
+         (-6.012818500554211e+249-1.926943776932387e+250j),
+         5e-13,),
+        ((-237.71710702931668+13.992648136816397j),
+         (-8.823803086106129e+281-5.009074181335139e+281j),
+         1e-13,),
+        ((-217.40285743524163-16.147667799398363j),
+         (-5.111612904844256e+251-4.907132127666742e+250j),
+         5e-13,),
+        ((-237.71710702931668-16.147667799398363j),
+         (-1.3256112779883167e+283-2.253002003455494e+283j),
+         5e-13,),
+    ],
+)
+def test_riemann_zeta_complex_avoid_underflow(z, desired, rtol):
     assert_allclose(sc.zeta(z), desired, rtol=rtol)

